sdk/src/client/stronghold/mod.rs

```diff
@@ -182,6 +182,12 @@ impl StrongholdAdapterBuilder {
     /// [`password()`]: Self::password()
     /// [`timeout()`]: Self::timeout()
     pub fn build<P: AsRef<Path>>(self, snapshot_path: P) -> Result<StrongholdAdapter, Error> {
+        if snapshot_path.as_ref().is_dir() {
+            // TODO: Add Error in 2.0 as its breaking.
+            // Issue #1197
+            panic!("Path is not a file: {:?}", snapshot_path.as_ref().to_path_buf());
```

Comment:
Let's return an IO error instead

Reply:
i dont have an std::io error. in https://github.com/iotaledger/iota-sdk/pull/1187/commits/482fe51b9063e5672527bc8aa8d255625b637fd6 i introduced an error type but that was breaking

Reply:
Not sure what you mean, just return `std::io::Error`? I just tested it it's fine.

Reply:
how? :D

`return Err(Error::Io(std::io::Error(format!("Path is not a file: {:?}", snapshot_path.unwrap()))));` doesnt work obviously